scour/scour.py

```diff
@@ -3178,20 +3178,10 @@ def remapNamespacePrefix(node, oldprefix, newprefix):
         remapNamespacePrefix(child, oldprefix, newprefix)
 
 
-def makeWellFormed(str):
-    # Don't escape quotation marks for now as they are fine in text nodes
-    # as well as in attributes if used reciprocally
-    #     xml_ents = { '<':'&lt;', '>':'&gt;', '&':'&amp;', "'":'&apos;', '"':'&quot;'}
+def makeWellFormed(str, quote=''):
```

Comment:
Could we put the new replacement logic below (in `serializeXML()`) into this function?

That way we'd keep all related functionality in a single function, a simple boolean argument (e.g. `escape_quotes`) could control behavior for attribute values vs. text nodes (i.e. disable the functionality in the latter case).

The type of quotes required could then be returned by the function.

Let me know what you think...

Edit: Since this would require some additional changes, let's complete this PR first and think about that suggestion later.... (Your input is still welcome; Let me know if you're interested in working on it)

Reply:
Certainly, that can be done, but my initial inquiry into the matter has led me to the conclusion that the result will actually be messier, mainly due to the way the unit tests will have to be written.

I think the resulting ugliness stems from the fact that `makeWellFormed` has no business choosing how to serialize a quoted string; using either single quotes or double quotes makes no difference to whether the result will be well formed&mdash;the choice should be the purview of a higher-level of abstraction, namely `serializeXML`, which makes the choice based on the criterion that the serialization should be as short as possible.

Also, what if an even higher level of abstraction (*e.g.*, the user) wants to ensure that every quoted string is delimited by double quotes, regardless of the length criterion? As it is now, it would be a simple matter of just telling `makeWellFormed` that a double quote is going to be used, and thereby letting `makeWellFormed` produced a well-formed value based on that fact.